src/HttpClientMock/RealRequest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brainbits\FunctionalTestHelpers\HttpClientMock;
+
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function http_build_query;
+use function in_array;
+use function strtolower;
+use function trim;
+
+use const PHP_EOL;
+
+final class RealRequest
+{
+    /**
+     * @param array<string, string>                                                                      $headers
+     * @param mixed[]                                                                                    $json
+     * @param array<string, string>                                                                      $queryParams
+     * @param array<string, string>                                                                      $requestParams
+     * @param array<string, array{name: string, filename?: string, mimetype?: string, content?: string}> $multiparts
+     */
+    public function __construct(
+        private string $method,
+        private string $uri,
+        private array $headers,
+        private string|null $content,
+        private array|null $json,
+        private array $queryParams,
+        private array $requestParams,
+        private array $multiparts,
+    ) {
+    }
+
+    public function getMethod(): string
+    {
+        return $this->method;
+    }
+
+    public function getUri(): string
+    {
+        return $this->uri;
+    }
+
+    public function hasHeader(string $key): bool
+    {
+        return in_array($key, array_map('strtolower', array_keys($this->headers)));
```

Comment:
It may be more error safe to normalize header keys in the constructor so that they cannot be forgotten.

Reply:
done. only did it this way so the RealRequest toString call would print the headers in the same way they were provided. but it's safer this way.